tests/integration/test_task_pool.py

```diff
@@ -1243,6 +1245,144 @@ async def test_detect_incomplete_tasks(
+                'allow implicit tasks': 'True',
+                'cycle point format': 'CCYY',
+            },
+            'scheduling': {
+                'initial cycle point': '0001',
+                'runahead limit': 'P3Y',
+                'graph': {
+                    'P1Y': 'a'
+                },
+            }
+        }
+        point = ISO8601Point
+
+    monkeypatch.setattr(
+        'cylc.flow.flags.cylc7_back_compat',
+        compat_mode == 'compat-mode',
+    )
+
+    id_ = flow(config)
+    schd = scheduler(id_)
+    async with start(schd):
+        schd.pool.compute_runahead(force=True)
+        assert int(str(schd.pool.runahead_limit_point)) == 4
+
+        # ensure task states are initiated with is_runahead status
+        assert schd.pool.get_task(point('0001'), 'a').state(is_runahead=False)
+        assert schd.pool.get_task(point('0005'), 'a').state(is_runahead=True)
+
+        # mark the first three cycles as running
+        for cycle in range(1, 4):
+            schd.pool.get_task(point(f'{cycle:04}'), 'a').state.reset(
+                TASK_STATUS_RUNNING
+            )
+
+        schd.pool.compute_runahead(force=True)
+        assert int(str(schd.pool.runahead_limit_point)) == 4  # no change
+
+        # mark cycle 1 as incomplete (but finished)
+        schd.pool.get_task(point('0001'), 'a').state.reset(
+            TASK_STATUS_SUBMIT_FAILED
+        )
+
+        schd.pool.compute_runahead(force=True)
+
+        if compat_mode == 'compat-mode':
+            # Cylc 7 does not count failed tasks in runahead computation.
+            assert int(str(schd.pool.runahead_limit_point)) == 5
+        else:
+            # Cylc 8 does count failed tasks in runahead computation.
+            assert int(str(schd.pool.runahead_limit_point)) == 4  # no change
+
+        # mark cycle 1 as complete
+        # (via task message so the task gets removed before runahead compute)
+        schd.task_events_mgr.process_message(
+            schd.pool.get_task(point('0001'), 'a'),
+            logging.INFO,
+            TASK_OUTPUT_SUCCEEDED
+        )
+        schd.pool.compute_runahead(force=True)
+        assert int(str(schd.pool.runahead_limit_point)) == 5  # +1
+
+
+@pytest.mark.parametrize(
+    'rhlimit', ['P2D', 'P2']
+)
+async def test_runahead_future_trigger(
+    flow,
+    scheduler,
+    start,
+    rhlimit
+):
+    """Equivalent time interval and cycle count runahead limits should yield
+    the same limit point, even if there is a future trigger.
+
+    See https://github.com/cylc/cylc-flow/pull/5893
+    """
+    id_ = flow({
+        'scheduler': {
+            'allow implicit tasks': 'True',
+            'cycle point format': 'CCYYMMDD',
+        },
+        'scheduling': {
+            'initial cycle point': '2001',
+            'runahead limit': rhlimit,
+            'graph': {
+                'P1D': 'a\na[+P1D] => b',
+            },
+        }
+    })
+    schd = scheduler(id_,)
+    async with start(schd):
```

Comment:
Might be useful to do this so that if the test fails, we can see what runahead debug log messages were produced

```suggestion
    async with start(schd, level=logging.DEBUG):
```